Update lightning-network-error.md

tlindi opened the pull request:
add information of nat=false when UPnP routers don't work.

## Description

nat=false is many times coming up on Telegram channels. It is also only option if one wants to run clearnet node and route UPnP is not co-operative / compatible.

Here's one example: https://github.com/mynodebtc/mynode/issues/840

## Checklist

* [X] mentioned related issue

tehelsper commented:
I had considered adding something like this, but I did not want to encourage custom configs since it leads to more issues. Once a config is custom, automatic updates and other settings stop working because the custom config overrides them.

tlindi commented:
Correct.

Maybe then just ADD switches under both bitcoind and Lightning configuration?

I'd love them too cause My current router is not compatible anymore wihh UPnP of those.

tehelsper commented:
If you set nat=false do you also need to specify your external IP? I thought that would also require a config being updated frequently and didn't want to support it outside of the already "expert mode" config where you can fully customize it. If it doesn't require the external ip though, I could add a toggle.

tlindi commented:
> If you set nat=false do you also need to specify your external IP?
>
Yes, or eg DynDNS DNS name. 

>I thought that would also require a config being updated frequently 
> 
wouldn't. If one uses DDNS inside of router/modem.

>and didn't want to support it outside of the already "expert mode" 
>config where you can fully customize it.
>
--and in addition to extra_conf extraconf lines
```
[Application Options]
tlsextradomain=yourfancydnsname.ddns.net
externalhosts=yourfancydnsname.ddns.net:9735
```

>If it doesn't require the external ip though, I could add a toggle.
>
So unfor it does need :-/

To use any apps on clearnet IRL one needs to 
- generate nginx conf HTTPS for public DNS name , 
- get non-selfgenerated SSL Certificate (certbot),
- open clearnet http to certbot, but prevent it to expose MyNode http pages and
- and few other thinks too :-/

I tried Caddy for those, but it cant be stopped to listen 80. So it breaks every mynode* related site :-/

too hard to get eg BTCPay or LNbits running on clearnet, for many...

tehelsper commented:
Yeah, that's similar to what I thought. I didn't want to add a toggle that would break most people when they don't understand all the other options necessary to configure.

Can you add nat=false to the additional config as well? It might override the original setting, but I have not tried it.

tlindi commented:
>Can you add nat=false to the additional config as well?
>It might override the original setting, but I have not tried it.
>
Yes, it does override, as others setting too , what ever I have needed (debug=.. tslextradomanin etc)

I have newer been in need to modify “standard” conf of LND and bitcoin since this “extra” was implemented :-D
T.mi

From: Taylor Helsper ***@***.***>
Sent: keskiviikko 2. huhtikuuta 2025 6.53
To: mynodebtc/mynode_docs ***@***.***>
Cc: Tomi Lind ***@***.***>; Author ***@***.***>
Subject: Re: [mynodebtc/mynode_docs] Update lightning-network-error.md (PR #62)


Yeah, that's similar to what I thought. I didn't want to add a toggle that would break most people when they don't understand all the other options necessary to configure.

Can you add nat=false to the additional config as well? It might override the original setting, but I have not tried it.

—
Reply to this email directly, view it on GitHub<https://github.com/mynodebtc/mynode_docs/pull/62#issuecomment-2771273503>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABXJ6Z25OGZ3Y566ECE34HL2XNNJZAVCNFSM6AAAAABZ6TCJCWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZRGI3TGNJQGM>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>
[tehelsper]tehelsper left a comment (mynodebtc/mynode_docs#62)<https://github.com/mynodebtc/mynode_docs/pull/62#issuecomment-2771273503>

Yeah, that's similar to what I thought. I didn't want to add a toggle that would break most people when they don't understand all the other options necessary to configure.

Can you add nat=false to the additional config as well? It might override the original setting, but I have not tried it.

—
Reply to this email directly, view it on GitHub<https://github.com/mynodebtc/mynode_docs/pull/62#issuecomment-2771273503>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABXJ6Z25OGZ3Y566ECE34HL2XNNJZAVCNFSM6AAAAABZ6TCJCWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZRGI3TGNJQGM>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

tehelsper commented:
OK, awesome. Would you mind adding info about the other config necessary? I'd be good with having a simple Tor-based solution and an Advanced IP solution using the extra config section.

tlindi commented:
> If you set nat=false do you also need to specify your external IP? 
>
Thread_functions.py seems to already get public IP. Maybe that could be proposed as configuration value to lnd.conf nat=false guys ,-) or it could update conf and, yes boot always when IP changes. But it would be automatic... I keep on dreaming ;-)

tehelsper commented:
Yeah, I could, but I _really_ don't want to change LND config based on the current IP. The last found IP can either be out of date or cause failures if the IP can't be found properly. There was a recent issue where one of the IP lookup locations was failing.

I wasn't clear - I just meant the Tor solution already described could be called "simple" since it would not require additional config. The IP solution with addition config would have "advanced" in the title just to encourage users to go with the current Tor based solution while also still explaining to more technical users that an IP-based solution exists.

tlindi commented:
> Yeah, I could, but I _really_ don't want to change LND config based on the current IP. The last found IP can either be out of date or cause failures if the IP can't be found properly. There was a recent issue where one of the IP lookup locations was failing.
> 
> I wasn't clear - I just meant the Tor solution already described could be called "simple" since it would not require additional config. The IP solution with addition config would have "advanced" in the title just to encourage users to go with the current Tor based solution while also still explaining to more technical users that an IP-based solution exists.

There you go. Is it Advanced enough to scare everyone to even try ;-) ?

tlindi commented:
Thank you appreciating my docs.

And believe me - from all the bundle nodes I have tried, this is not the most easy way to get onto clearnet, but only one I have found!

tlindi commented:
Ackzyally... There's https_public_services nginx conf and ports 600x catching of apps. 

I tested to clone it to catch Port 6000 for all public hostnames needed. And added certbot. And I get with two router Port forwardings _any_ App to be available from clearnet via https.

Would there be a chance to get PR adding certbot and modified nginx ever merged to main?

If this same advanced (or Additional "Even More Advanced") document would be created to explain usage?

@tehelsper ?

tehelsper commented:
Maybe partially. I woudln't mind if the apps were simple to maintain and installed to be available for advanced users. I don't think I would make it something available via the UI to just prevent less experienced users from breaking things.

The public apps idea is probably not going to happen. It was an idea to make public access to things like lnbits or btcpay simple through a subdomain like user.lnbits.mynodebtc.com. It somewhat worked, but all traffic was getting proxied and was a complex setup. I should remove those nginx files or at least just comment them out.

tlindi commented:
> I should remove those nginx files or at least just comment them out.
>
Yes. Even I couldn't figure out how they were supposed to work. Just leave port 6000 for `# Advanced Users Clearnet https in` and I'd be quite happy.

As I have time I can PR something really minimal that could help advanced users, and not enable others too easy to wreck their node.

Anyhow whole certbot, nginx & ufw area is for users with good or excellent networking skills. 

Could we keep this issue open, so it is easier for me to return to topic in the future?
